static/app/components/events/eventStatisticalDetector/eventComparison/eventDisplay.tsx

```diff
@@ -0,0 +1,226 @@
+import {useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {CompactSelect} from 'sentry/components/compactSelect';
+import EmptyStateWarning from 'sentry/components/emptyStateWarning';
+import {MINIMAP_HEIGHT} from 'sentry/components/events/interfaces/spans/constants';
+import {ActualMinimap} from 'sentry/components/events/interfaces/spans/header';
+import WaterfallModel from 'sentry/components/events/interfaces/spans/waterfallModel';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import TextOverflow from 'sentry/components/textOverflow';
+import {IconEllipsis, IconJson, IconLink} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {EventTransaction, Project} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {useDiscoverQuery} from 'sentry/utils/discover/discoverQuery';
+import EventView from 'sentry/utils/discover/eventView';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
+import {getShortEventId} from 'sentry/utils/events';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+
+// A hook for getting "sample events" for a transaction
+// In its current state it will just fetch at most 5 events that match the
+// transaction name within a range of the duration baseline provided
+function useFetchSampleEvents({
+  start,
+  end,
+  transaction,
+  durationBaseline,
+  projectId,
+}: {
+  durationBaseline: number;
+  end: number;
+  projectId: number;
+  start: number;
+  transaction: string;
+}) {
+  const location = useLocation();
+  const organization = useOrganization();
+  const eventView = new EventView({
+    dataset: DiscoverDatasets.DISCOVER,
```

Comment:
minor: I think you can query against TRANSACTIONS directly since you know that's what you're looking for